src/modules/Hosts/HostsUILib/Helpers/HostsService.cs

```diff
@@ -128,11 +128,17 @@ public async Task WriteAsync(string additionalLines, IEnumerable<Entry> entries)
                 throw new NotRunningElevatedException();
             }
 
-            if (_fileSystem.FileInfo.FromFileName(HostsFilePath).IsReadOnly)
+            if (HasAttribute(FileAttributes.ReadOnly))
             {
                 throw new ReadOnlyHostsException();
             }
 
+            var hidden = HasAttribute(FileAttributes.Hidden);
+            if (hidden)
+            {
+                RemoveAttribute(FileAttributes.Hidden);
```

Comment:
should we do this? is this how VS Code handles that ?

Reply:
we can unhide it , modify it, hide it. But ideally there should be a better solution, just modify it even if it's hidden

Reply:
I investigated into the error and changed the approach instead of messing with attributes 🚀 
I have updated the PR detailed description.